app/controllers/storytime/dashboard/posts_controller.rb

```diff
@@ -53,11 +60,17 @@ def create
       def update
         authorize @post
         @post.draft_user_id = current_user.id
+        @post.assign_attributes(post_params)
+
+        if post_params["published"] == "0"
+          @post.published_at_date = nil
+          @post.published_at_time = nil
+        end
```

Comment:
Any reason to do this in the controller? Could we do this in the published setter in has_versions?

```
def published=(val)
  if val == "1"
    self.published_at_date = Time.now.to_date
    self.published_at_time = Time.now
  else
    self.published_at_date = nil
    self.published_at_time = nil
  end
end
```

Thoughts?